### PR TITLE
[Core/Notify] feat: Move NotifyConfigDTO to Explorer module

### PR DESCRIPTION
**Pull request by Elyniss**
Merge after #1189 

Introduces new Notification Type structure that additionally has `id` field for easier management. Required by specs

### CODE REVIEW

File: core/android/src/main/kotlin/com/walletconnect/android/internal/common/explorer/ExplorerRepository.kt [hunk 1]
```diff
@@ -67,6 +71,18 @@ class ExplorerRepository(
         }
     }
 
+    suspend fun getNotifyConfig(
+        appDomain: String,
```
Comment (jakubuid) [inline core/android/src/main/kotlin/com/walletconnect/android/internal/common/explorer/ExplorerRepository.kt:75]: what is addDomain?

File: protocol/notify/src/main/kotlin/com/walletconnect/notify/di/NotifyStorageModule.kt [hunk 2]
```diff
@@ -19,29 +19,40 @@ internal fun notifyStorageModule(dbName: String) = module {
...
+                    when (values.size) {
+                        // Backward compatibility with example 'name=description,isSubscribed'
+                        2 -> key.lowercase() to values.run {
+                            Triple(key, this.first().trim(), this.last().trim().toBoolean())
+                        }
+                        // Current compatibility with example 'id=name,description,isSubscribed'
+                        3 -> key.lowercase() to values.run {
+                            Triple(this.first().trim(), this[1].trim(), this.last().trim().toBoolean())
+                        }
+                        // Fail-over
```
Comment (jakubuid) [inline protocol/notify/src/main/kotlin/com/walletconnect/notify/di/NotifyStorageModule.kt:47]: Do we need all of these comments?

**Review by jakubuid — DISMISSED** [review]

File: core/android/src/main/kotlin/com/walletconnect/android/internal/common/explorer/ExplorerRepository.kt (hunk 1 above)
Comment (Elyniss) [reply core/android/src/main/kotlin/com/walletconnect/android/internal/common/explorer/ExplorerRepository.kt:75]: domain of an app that is registered in Cloud. I.e gm.walletconnect.com

**Review by kacperoak — APPROVED** [review]

**Comment by Elyniss** [comment]
Merging as PR was already approved. Changed base make approval disappear

**Pull request merged**